From 6d63e2d76401d2d7b86d62a07235321f858da8ea Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Fri, 26 May 2023 14:06:43 +0800
Subject: [PATCH] nnstreamer: Fix detect objects array is out of bound in
 tensordec-boundingbox

If length of detect objects array is zero, don't access array.

Signed-off-by: kidd-kw.chen <kidd-kw.chen@mediatek.com>
---
 ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c b/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c
index 28f98a9c..4bb70f9b 100644
--- a/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c
+++ b/ext/nnstreamer/tensor_decoder/tensordec-boundingbox.c
@@ -1284,6 +1284,9 @@ nms (GArray * results, gfloat threshold)
   }
 
   i = 0;
+  if (results->len == 0)
+    return;
+
   do {
     detectedObject *a = &g_array_index (results, detectedObject, i);
     if (a->valid == FALSE)
@@ -1291,7 +1294,6 @@ nms (GArray * results, gfloat threshold)
     else
       i++;
   } while (i < results->len);
-
 }
 
 /**
-- 
2.18.0

